pyanaconda/ui/gui/gk_keyboard_manager.py

```diff
@@ -0,0 +1,136 @@
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+
+from pyanaconda.core.signal import Signal
+from pyanaconda.keyboard import join_layout_variant, parse_layout_variant, KeyboardConfigError
+from pyanaconda.modules.common.constants.services import GK_INPUT_SOURCES
+
+
+class GkKeyboardManager(object):
+    """Class wrapping GNOME Kiosk's input sources API."""
+
+    def __init__(self):
+        self.compositor_selected_layout_changed = Signal()
+        self.compositor_layouts_changed = Signal()
+
+        object_path = GK_INPUT_SOURCES.object_path + '/InputSources/Manager'
+        self._proxy = GK_INPUT_SOURCES.get_proxy(object_path=object_path)
+        self._proxy.PropertiesChanged.connect(self._on_properties_changed)
+
+    def _on_properties_changed(self, interface, changed_props, invalid_props):
+        for prop in changed_props:
+            if prop == 'SelectedInputSource':
+                self.compositor_selected_layout_changed.emit()
+            if prop == 'InputSources':
+                self.compositor_layouts_changed.emit()
+
+    def _path_to_layout(self, layout_path):
+        """
+        Transforms a layout path as returned by GNOME Kiosk to "layout (variant)".
+
+        :param layout_path: D-Bus path to the layout.
+            (e.g. "/org/gnome/Kiosk/InputSources/xkb_cz_2b_mon_5f_todo_5f_galik")
+        :type layout_path: str
+        :return: The layout with format "layout (variant)" (e.g. "cn (mon_todo_galik)")
+        :rtype: str
+
+        :raise KeyboardConfigError: if layouts with invalid backend type is found
+        """
+
+        layout_proxy = GK_INPUT_SOURCES.get_proxy(object_path=layout_path)
+
+        if layout_proxy.BackendType != 'xkb':
+            raise KeyboardConfigError('Failed to get configuration from compositor')
+
+        if '+' in layout_proxy.BackendId:
+            layout, variant = layout_proxy.BackendId.split('+')
+            return join_layout_variant(layout, variant)
+        else:
+            return layout_proxy.BackendId
+
+    def _layout_to_xkb(self, layout_variant):
+        """
+        Transforms a "layout (variant)" to a "('xkb', 'layout+variant')".
+
+        :param layout_variant: The layout with format "layout (variant)" (e.g. "cz (qwerty)")
+        :type layout_variant: str
+        :return: The layout with format "('xkb', 'layout+variant')" (e.g. "('xkb', 'cz+qwerty')")
+        :rtype: str
+        """
+
+        layout, variant = parse_layout_variant(layout_variant)
+        if variant:
+            return ('xkb', "%s+%s" % (layout, variant))
```

Comment:
For string formatting solution please use `.format` as mentioned in https://anaconda-installer.readthedocs.io/en/latest/contributing.html#code-conventions